Lib/pathlib.py

```diff
@@ -1385,6 +1385,132 @@ def expanduser(self):
 
         return self
 
+    def walk(self, on_error=None, follow_links=False):
+        """Generate a top-down directory tree from this directory
+
+        For each directory in the directory tree rooted at self (including
+        self but excluding '.' and '..'), yields a 3-tuple
+
+        root_directory, child_directory_names, child_file_names
+
+        The caller can modify the child_directory_names list in-place
+        (e.g., via del or slice assignment), and walk will only recurse into
+        the subdirectories whose names remain in child_directory_names; this
+        can be used to prune the search, or to impose a specific order of
+        visiting.
+
+        By default errors from Path._scandir() call are ignored.  If
+        optional arg 'on_error' is specified, it should be a function; it
+        will be called with one argument, an OSError instance.  It can
+        report the error to continue with the walk, or raise the exception
+        to abort the walk.  Note that the filename is available as the
+        filename attribute of the exception object.
+
+        By default, Path.walk does not follow symbolic links to subdirectories
+        on systems that support them.  In order to get this functionality, set
+        the optional argument 'follow_links' to true.
+
+        Caution: if self is a relative Path, don't change the
+        current working directory between resumptions of walk. walk never
+        changes the current directory, and assumes that the client doesn't
+        either.
+
+        Example:
+
+        from pathlib import Path
+        for root, dirs, files in Path('python/Lib/email'):
+            print(root, "consumes", end="")
+            print(sum((root / file).stat().st_size for file in files), end="")
+            print("bytes in", len(files), "non-directory files")
+            if 'CVS' in dirs:
+                dirs.remove('CVS')  # don't visit CVS directories
```

Comment:
This example is quite old now!  Could be `".git"` to make more sense to modern readers.  Or `'__pycache__'`!